src/main/java/com/aws/greengrass/shadowmanager/sync/model/FullShadowSyncRequest.java

```diff
@@ -68,8 +118,25 @@ public boolean isUpdateNecessary(SyncContext context) {
      * @throws InterruptedException     if the thread is interrupted while syncing shadow with cloud.
      */
     @Override
-    public void execute(SyncContext context) throws RetryableException, SkipSyncRequestException, InterruptedException {
+    public void execute(SyncContext context) throws RetryableException, SkipSyncRequestException,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 21. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 125-138 into a separate method.